models/repo_list.go

```diff
@@ -182,6 +185,10 @@ func SearchRepositoryByName(opts *SearchRepoOptions) (repos RepositoryList, coun
 		cond = cond.And(builder.Eq{"is_private": false})
 	}
 
+	if opts.OwnerID > 0 && opts.AllPublic {
+		cond = cond.Or(builder.Eq{"is_private": false})
```

Comment:
I think this is a bug. If `opts.OwnerID > 0 && opts.AllPublic` is true, then we will search for all public repos, even those that do not match the keyword.

Reply:
Fixed